java-checks/src/main/java/org/sonar/java/checks/spring/StatusCodesOnResponseCheck.java

```diff
@@ -0,0 +1,222 @@
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S6863")
+public class StatusCodesOnResponseCheck extends IssuableSubscriptionVisitor {
+
+  public static final String RESPONSE_ENTITY = "org.springframework.http.ResponseEntity";
+  public static final String OK_ISSUE_MESSAGE = "Use the \"ResponseEntity.ok()\" method or set the status to \"HttpStatus.OK\".";
+  public static final String ERROR_ISSUE_MESSAGE = "Use the \"ResponseEntity.badRequest()\" or \"ResponseEntity.notFound()\" method" +
+    "or set the status to \"HttpStatus.INTERNAL_SERVER_ERROR\" or \"HttpStatus.NOT_FOUND\".";
+
+  private final List<String> okCodes = List.of("CONTINUE",
+    "SWITCHING_PROTOCOLS",
+    "PROCESSING",
+    "CHECKPOINT",
+    "OK",
+    "CREATED",
+    "ACCEPTED",
+    "NON_AUTHORITATIVE_INFORMATION",
+    "NO_CONTENT",
+    "RESET_CONTENT",
+    "PARTIAL_CONTENT",
+    "MULTI_STATUS",
+    "ALREADY_REPORTED",
+    "IM_USED",
+    "MULTIPLE_CHOICES",
+    "MOVED_PERMANENTLY",
+    "FOUND",
+    "SEE_OTHER",
+    "NOT_MODIFIED",
+    "TEMPORARY_REDIRECT",
+    "PERMANENT_REDIRECT");
+
+  private final List<String> errorCodes = List.of("BAD_REQUEST",
+    "UNAUTHORIZED",
+    "PAYMENT_REQUIRED",
+    "FORBIDDEN",
+    "NOT_FOUND",
+    "METHOD_NOT_ALLOWED",
+    "NOT_ACCEPTABLE",
+    "PROXY_AUTHENTICATION_REQUIRED",
+    "REQUEST_TIMEOUT",
+    "CONFLICT",
+    "GONE",
+    "LENGTH_REQUIRED",
+    "PRECONDITION_FAILED",
+    "PAYLOAD_TOO_LARGE",
+    "URI_TOO_LONG",
+    "UNSUPPORTED_MEDIA_TYPE",
+    "REQUESTED_RANGE_NOT_SATISFIABLE",
+    "EXPECTATION_FAILED",
+    "I_AM_A_TEAPOT",
+    "UNPROCESSABLE_ENTITY",
+    "LOCKED",
+    "FAILED_DEPENDENCY",
+    "UPGRADE_REQUIRED",
+    "PRECONDITION_REQUIRED",
+    "TOO_MANY_REQUESTS",
+    "REQUEST_HEADER_FIELDS_TOO_LARGE",
+    "UNAVAILABLE_FOR_LEGAL_REASONS",
+    "INTERNAL_SERVER_ERROR",
+    "NOT_IMPLEMENTED",
+    "BAD_GATEWAY",
+    "SERVICE_UNAVAILABLE",
+    "GATEWAY_TIMEOUT",
+    "HTTP_VERSION_NOT_SUPPORTED",
+    "VARIANT_ALSO_NEGOTIATES",
+    "INSUFFICIENT_STORAGE",
+    "LOOP_DETECTED",
+    "BANDWIDTH_LIMIT_EXCEEDED",
+    "NOT_EXTENDED",
+    "NETWORK_AUTHENTICATION_REQUIRED");
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return List.of(Tree.Kind.CLASS);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    ClassTree classTree = (ClassTree) tree;
+    if (!isClassController(classTree)) {
+      return;
+    }
+
+    MethodInvocationVisitor methodInvocationVisitor = new MethodInvocationVisitor();
+    classTree.accept(methodInvocationVisitor);
+  }
+
+  private class MethodInvocationVisitor extends BaseTreeVisitor {
+
+    MethodMatchers statusMethodMatchers = MethodMatchers.create()
```

Comment:
We should mark this matcher and the following ones as `static final`. Perhaps `private` as well?